cluster-topology.md

```diff
@@ -0,0 +1,273 @@
+```
+
+We'll reference the above "`default-worker`" class in [an updated Cluster configuration that declares a second worker pool](yamls/clusters/docker-cluster-one-second-worker-pool.yaml).
+
+The difference between this spec and the original spec used to create the "`docker-cluster-one`" Cluster:
+
+```sh
+diff yamls/clusters/docker-cluster-one.yaml yamls/clusters/docker-cluster-one-second-worker-pool.yaml
+```
+
+Output:
+
+```
+35a36,38
+>         - class: default-worker
+>           name: md-1
+>           replicas: 1
+```
+
+The above shows that we've added a new `MachineDeployment` called `md-1` (Our existing `MachineDeployment` is named `md-0`) to our new spec. By applying this modified spec to our kind management cluster we will initiate the creation of a new node, from this new node pool:
+
+```sh
+kubectl --kubeconfig kubecon-na-22-capi-lab.kubeconfig apply -f yamls/clusters/docker-cluster-one-second-worker-pool.yaml
+```
+
+Output:
+
+```
+cluster.cluster.x-k8s.io/docker-cluster-one configured
+
+Let's watch those new nodes come online:
+
+```sh
+kubectl --kubeconfig cluster-one.kubeconfig get nodes -w
+```
+
+Output:
+
+```
+NAME                                             STATUS   ROLES           AGE   VERSION
+docker-cluster-one-md-0-nrh7k-75ddc4778f-wwpg9   Ready    <none>          25h   v1.24.6
+docker-cluster-one-xbqb2-bcjvj                   Ready    control-plane   26h   v1.24.6
+docker-cluster-one-md-1-ksqwt-8489684d5b-fpgjs   NotReady                   <none>          0s    v1.24.6
+docker-cluster-one-md-1-ksqwt-8489684d5b-fpgjs   NotReady                   <none>          46s   v1.24.6
+docker-cluster-one-md-1-ksqwt-8489684d5b-fpgjs   NotReady                   <none>          46s   v1.24.6
+docker-cluster-one-md-1-ksqwt-8489684d5b-fpgjs   Ready                      <none>          76s   v1.24.6
+```
+
+## Remove a worker node
+
+We can now demonstrate how easy it is to "roll back" such a change, as well as show how to remove an existing node pool from your Cluster configuration. In our case it's as easy as reapplying the original Cluster spec, which only declares one "`md-0`" node pool:
+
+```sh
+kubectl --kubeconfig kubecon-na-22-capi-lab.kubeconfig apply -f yamls/clusters/docker-cluster-one.yaml
+```
+
+Once again, we should observe only one running worker node, in the "`md-0`" pool:
+
+```sh
+kubectl --kubeconfig cluster-one.kubeconfig get nodes
+```
+
+Output:
+
+```
+NAME                                            STATUS   ROLES           AGE    VERSION
+docker-cluster-one-md-0-lq4f8-b59497b9d-xchpm   Ready    <none>          99m    v1.24.6
+docker-cluster-one-mvthd-k7dwf                  Ready    control-plane   174m   v1.24.6
+```
+
+## MachineHealthChecks and Remediation
+
+You are now in control of your Cluster's topology configuration! [Let's next explore MachineHealthChecks and Remediation to configure operational self-healing](machine-health-checks-remediation.md)...
```

Comment:
@sbueringer FYI, when we actually do the MachineHealthCheck stuff we'll want to circle back and make sure this link is valid (i.e., you might use a different name for the `.md` file...)

Reply:
+1

I've added a sub-task that we should check the chaining in general